includes/wccom-site/class-wc-wccom-site-installer.php

```diff
@@ -293,6 +293,7 @@ private static function get_product_info( $product_id ) {
 		} else {
 			// Retrieve download URL for non-wporg product.
 			WC_Helper_Updater::flush_updates_cache();
+			WC_Helper::_flush_subscriptions_cache();
```

Comment:
This is too late - the previous `elseif` is where we check the subscription - see `WC_Helper::has_product_subscription( $product_id )` above.

I'd suggest that we either move this earlier in the function, or explictly clear all the helper caches at the beginning the `POST install` (start) API handling.

Reply:
Indeed, @haszari, I know. Sorry for having `needs review` status. Actually code was in progress. I chose clear the cache after figuring out whether it's WP.org product or not.